server/src/resolvers/Message/subscription.ts

```diff
@@ -25,10 +25,7 @@ export const onMessage = subscriptionField('onMessage', {
         },
       });
 
-      const correctChannel = !!membership; // Message is sent to the auth user's channel.
-      const fromSelf = payload.senderId === userId; // Message is sent by the auth user.
-
-      return correctChannel && !fromSelf;
```

Comment:
I think we should filter the `sent` user device token. If user A has an iPhone and Android, and the user has sent with an Android device, the user should not receive the message from `Android`.

Reply:
`payload` here does not have device token, so I don't think we can detect which device a message is sent from. Is there a way to find the origin device of a message?

Reply:
We are currently collecting user's push tokens and sending notifications via that. I think we need to add a payload to filter the token 🤔

If it looks a bit complicated, we may talk about it in the next `PR`. I'll make a proposal :)

Reply:
Actually, I was concerned about the notification. Sorry for the confusion. This `PR` just looks great!